ts/components/conversation/CreateGroupDialog.tsx

```diff
@@ -0,0 +1,212 @@
+    this.onKeyUp = this.onKeyUp.bind(this);
+    this.closeDialog = this.closeDialog.bind(this);
+    this.onGroupNameChanged = this.onGroupNameChanged.bind(this);
+
+    let friends = this.props.friendList;
+    friends = friends.map(d => {
+      const lokiProfile = d.getLokiProfile();
+      const name = lokiProfile ? lokiProfile.displayName : 'Anonymous';
+
+      return {
+        id: d.id,
+        authorPhoneNumber: d.id,
+        authorProfileName: name,
+        selected: false,
+        authorName: name, // different from ProfileName?
+        authorColor: d.getColor(),
+        checkmarked: false,
+      };
+    });
+
+    this.state = {
+      friendList: friends,
+      groupName: '',
+      errorDisplayed: false,
+      // if empty, the initial height is 0, which is not desirable
+      errorMessage: 'placeholder',
+    };
+
+    window.addEventListener('keyup', this.onKeyUp);
+  }
+
+  public onClickOK() {
+    const members = this.state.friendList
+      .filter(d => d.checkmarked)
+      .map(d => d.id);
+
+    if (!this.state.groupName.trim()) {
+      this.onShowError(this.props.i18n('emptyGroupNameError'));
+
+      return;
+    }
+
+    window.doCreateGroup(this.state.groupName, members);
+
+    this.closeDialog();
+  }
+
+  public render() {
+    const checkMarkedCount = this.getMemberCount();
+
+    const titleText = `${this.props.titleText} (Members: ${checkMarkedCount})`;
+    const okText = this.props.okText;
+    const cancelText = this.props.cancelText;
+
+    const errorMessageClasses = classNames(
+      'error-message',
+      this.state.errorDisplayed ? 'error-shown' : 'error-faded'
+    );
+
+    return (
+      <div className="content">
+        <p className="titleText">{titleText}</p>
+        <p className={errorMessageClasses}>{this.state.errorMessage}</p>
+        <input
+          type="text"
+          id="group-name"
+          className="group-name"
+          placeholder="Group Name"
+          value={this.state.groupName}
+          onChange={this.onGroupNameChanged}
+          tabIndex={0}
+          required={true}
+          autoFocus={true}
+          aria-required={true}
+        />
+        <div className="friend-selection-list">
+          <MemberList
+            members={this.state.friendList}
+            selected={{}}
+            i18n={this.props.i18n}
+            onMemberClicked={this.onMemberClicked}
+          />
+        </div>
+        <div className="buttons">
+          <button className="cancel" tabIndex={0} onClick={this.closeDialog}>
+            {cancelText}
+          </button>
+          <button className="ok" tabIndex={0} onClick={this.onClickOK}>
+            {okText}
+          </button>
+        </div>
+      </div>
+    );
+  }
+
+  private onShowError(msg: string) {
+    if (this.state.errorDisplayed) {
+      return;
+    }
+
+    this.setState({
+      errorDisplayed: true,
+      errorMessage: msg,
+    });
+
+    setTimeout(() => {
```

Comment:
Probably not a big deal but this won't ensure that the errors will be shown for 3 seconds if you have multiple errors in a row faster than 3s

Reply:
Won't 
>if (this.state.errorDisplayed) {
      return;
    }

above take care of that?

Reply:
Ah yes true but then that means you won't see any new errors that might be different from the old errors?

Reply:
Also true, but I think this is unlikely to happen. If this becomes an issue I guess we could check if the error is different and show it.